packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -61,11 +63,30 @@ function useBlockEditorSettings( settings, hasTemplate ) {
 			),
 			hasResolvedLocalSiteData: hasFinishedResolvingSiteData,
 			baseUrl: siteData?.url || '',
+			userCanCreatePages: canUser( 'create', 'pages' ),
 		};
 	}, [] );
 
 	const { undo } = useDispatch( editorStore );
 
+	const { saveEntityRecord } = useDispatch( coreStore );
+
+	/**
+	 * Creates a Post entity.
+	 * This is utilised by the Link UI to allow for on-the-fly creation of Posts/Pages.
+	 *
+	 * @param {Object} options parameters for the post being created. These mirror those used on 3rd param of saveEntityRecord.
+	 * @return {Object} the post type object that was created.
+	 */
+	const createPageEntity = ( options ) => {
+		if ( ! userCanCreatePages ) {
+			return Promise.reject( {
+				message: __( 'You do not have permission to create Pages.' ),
+			} );
+		}
```

Comment:
Minor: Do we need this? It seems like there won't even be an UI for creating pages if the user doesn't have the permission, and we should probably just delegate the error message in the API response. WDYT?

Reply:
I don't think we _need_ it because as you say the UI should be disabled. However, if there to be some bug there, I'd like a clear and precise error message to be displayed to the user. Otherwise we get a rubbish default.

Reply:
Just saying this because we have to be careful about developer-only error messages in packages, they can't be minified. That's why I think these kinds of messages should be in the API responses. I'm just nitpicking on best practices though, I'm sure it doesn't matter that much in this context 😅.

Reply:
>  developer-only error messages

That's just it, this isn't developer only. This error message would be directly exposed to the user in the LinkControl in the scenario that they were accidentally able to trigger the `Create page` UI. It provides clear feedback that this action is disallowed.

It's a fail safe.

Does that help?